packages/react-data-grid/src/common/editors/EditorContainer.tsx

```diff
@@ -56,50 +63,48 @@ export default class EditorContainer extends React.Component {
     }
   }
 
-  isKeyExplicitlyHandled = (key) => {
-    return isFunction(this[`onPress${key}`]);
-  };
-
-  checkAndCall = (methodName, args) => {
-    if (isFunction(this[methodName])) {
-      this[methodName](args);
+  onKeyDown = (e: KeyboardEvent<HTMLElement>) => {
+    switch (e.key) {
+      case 'Enter':
+        this.onPressEnter();
+        break;
+      case 'Tab':
+        this.onPressTab();
+        break;
+      case 'Escape':
+        this.onPressEscape(e);
+        break;
+      case 'ArrowDown':
+        this.onPressArrowDown(e);
+        break;
+      case 'ArrowUp':
+        this.onPressArrowUp(e);
+        break;
+      case 'ArrowLeft':
+        this.onPressArrowLeft(e);
+        break;
+      case 'ArrowRight':
+        this.onPressArrowRight(e);
+        break;
+      default:
+        break;
     }
-  };
 
-  onKeyDown = (e) => {
-    if (isCtrlKeyHeldDown(e)) {
-      this.checkAndCall('onPressKeyWithCtrl', e);
-    } else if (this.isKeyExplicitlyHandled(e.key)) {
-      // break up individual keyPress events to have their own specific callbacks
-      const callBack = `onPress${e.key}`;
-      this.checkAndCall(callBack, e);
-    } else if (isKeyPrintable(e.keyCode)) {
-      e.stopPropagation();
-      this.checkAndCall('onPressChar', e);
-    }
-
-    // Track which keys are currently down for shift clicking etc
-    this._keysDown = this._keysDown || {};
-    this._keysDown[e.keyCode] = true;
-    if (isFunction(this.props.onGridKeyDown)) {
+    if (this.props.onGridKeyDown) {
       this.props.onGridKeyDown(e);
     }
   };
 
-  setEditorRef = (editor) => {
-    this.editor = editor;
-  }
-
-  createEditor = () => {
-    const editorProps = {
-      ref: this.setEditorRef,
+  createEditor() {
+    const editorProps: EditorProps = {
```

Comment:
Maybe we should use `PropsWithRef<EditorProps>`.
That way we can remove the `ref` property on `EditorProps`, and it'll be re-usable for other components.